File: app/src/main/java/com/example/botcoin/za/co/botcoin/wallet/fragments/WalletMenuFrag.java
package com.example.botcoin.za.co.botcoin.wallet.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.LinearLayoutCompat;
import androidx.fragment.app.Fragment;
import com.example.botcoin.MainActivity;
import com.example.botcoin.R;
import com.example.botcoin.za.co.botcoin.utils.FragmentUtils;
import com.example.botcoin.za.co.botcoin.wallet.fragments.wallet.menu.fragment.ReceiveFrag;
import com.example.botcoin.za.co.botcoin.wallet.fragments.wallet.menu.fragment.SendFrag;

public class WalletMenuFrag extends Fragment
{
    private LinearLayoutCompat receiveOption;
    private LinearLayoutCompat sendOption;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.frag_wallet_menu, container, false);

        addReceiveListener(view);
        addSendListener(view);

        return view;
    }

    private void addSendListener(View view)
    {
        this.sendOption = view.findViewById(R.id.linearLayoutSendOption);
        this.sendOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                bundle.putString("asset", getArguments().getString("asset"));

                SendFrag sendFrag = new SendFrag();
                sendFrag.setArguments(bundle);
                FragmentUtils.startFragment(((MainActivity)getActivity()).getSupportFragmentManager(), sendFrag, R.id.fragContainer, ((MainActivity)getActivity()).getSupportActionBar(), "Wallet Menu",true, false, true, null);
            }
        });
    }

    private void addReceiveListener(View view)
    {
        this.receiveOption = view.findViewById(R.id.linearLayoutReceiveOption);
        this.receiveOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                bundle.putString("asset", getArguments().getString("asset"));

                ReceiveFrag receiveFrag = new ReceiveFrag();
                receiveFrag.setArguments(bundle);
                FragmentUtils.startFragment(((MainActivity)getActivity()).getSupportFragmentManager(), receiveFrag, R.id.fragContainer, ((MainActivity)getActivity()).getSupportActionBar(), "Wallet Menu",true, false, true, null);
            }
        });
    }
}

File: app/src/main/java/za/co/botcoin/view/wallet/WalletMenuFrag.kt
package za.co.botcoin.view.wallet

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation
import za.co.botcoin.R
import za.co.botcoin.databinding.WalletMenuFragmentBinding
import za.co.botcoin.view.wallet.menu.OrdersFrag
import za.co.botcoin.view.wallet.menu.ReceiveFrag
import za.co.botcoin.view.wallet.menu.SendFrag

class WalletMenuFrag : Fragment(R.layout.wallet_menu_fragment) {
    private lateinit var binding: WalletMenuFragmentBinding

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        this.binding = WalletMenuFragmentBinding.bind(view)

        arguments?.let {
            addReceiveListener(it.getString("asset") ?: "")
            addSendListener(it.getString("asset") ?: "")
            addOrdersListener(it.getString("asset") ?: "")
        }
    }

    private fun addSendListener(asset: String) {
        this.binding.linearLayoutSendOption.setOnClickListener {
            val action = WalletMenuFragDirections.actionWalletMenuFragToSendFrag(asset)
            Navigation.findNavController(it).navigate(action)
        }
    }

    private fun addReceiveListener(asset: String) {
        this.binding.linearLayoutReceiveOption.setOnClickListener {
            val action = WalletMenuFragDirections.actionWalletMenuFragToReceiveFrag(asset)
            Navigation.findNavController(it).navigate(action)
        }
    }

    private fun addOrdersListener(asset: String) {
        this.binding.linearLayoutOrdersOption.setOnClickListener {
           val action = WalletMenuFragDirections.actionWalletMenuFragToOrdersFrag(asset)
            Navigation.findNavController(it).navigate(action)
        }
    }
}